com.unity.ml-agents/Tests/Editor/ParameterLoaderTest.cs

```diff
@@ -0,0 +1,260 @@
+            Assert.AreEqual(0, errors.Count()); // There should not be any errors
+        }
+
+        [Test]
+        public void TestCheckModelThrowsVectorObservation1()
+        {
+            var model = ModelLoader.Load(continuous2vis8vec2actionModel);
+
+            var brainParameters = GetContinuous2vis8vec2actionBrainParameters();
+            brainParameters.vectorObservationSize = 9; // Invalid observation
+            var errors = BarracudaModelParamLoader.CheckModel(model, brainParameters, new SensorComponent[] { sensor_21_20_3, sensor_20_22_3 });
+            Assert.Greater(errors.Count(), 0);
+
+            brainParameters = GetContinuous2vis8vec2actionBrainParameters();
+            brainParameters.numStackedVectorObservations = 2;// Invalid stacking
+            errors = BarracudaModelParamLoader.CheckModel(model, brainParameters, new SensorComponent[] { sensor_21_20_3, sensor_20_22_3 });
+            Assert.Greater(errors.Count(), 0);
+        }
+
+        [Test]
+        public void TestCheckModelThrowsVectorObservation2()
+        {
+            var model = ModelLoader.Load(discrete1vis0vec_2_3action_recurrModel);
+
+            var brainParameters = GetDiscrete1vis0vec_2_3action_recurrModelBrainParameters();
+            brainParameters.vectorObservationSize = 1; // Invalid observation
+            var errors = BarracudaModelParamLoader.CheckModel(model, brainParameters, new SensorComponent[] { sensor_21_20_3 });
+            Assert.Greater(errors.Count(), 0);
+        }
+
+        [Test]
+        public void TestCheckModelThrowsAction1()
+        {
+            var model = ModelLoader.Load(continuous2vis8vec2actionModel);
+
+            var brainParameters = GetContinuous2vis8vec2actionBrainParameters();
+            brainParameters.vectorActionSize = new int[] { 3 }; // Invalid action
+            var errors = BarracudaModelParamLoader.CheckModel(model, brainParameters, new SensorComponent[] { sensor_21_20_3, sensor_20_22_3 });
+            Assert.Greater(errors.Count(), 0);
+
+            brainParameters = GetContinuous2vis8vec2actionBrainParameters();
+            brainParameters.vectorActionSpaceType = SpaceType.Discrete;// Invalid SpaceType
+            errors = BarracudaModelParamLoader.CheckModel(model, brainParameters, new SensorComponent[] { sensor_21_20_3, sensor_20_22_3 });
+            Assert.Greater(errors.Count(), 0);
+        }
+
+        [Test]
+        public void TestCheckModelThrowsAction2()
+        {
+            var model = ModelLoader.Load(discrete1vis0vec_2_3action_recurrModel);
+
+            var brainParameters = GetDiscrete1vis0vec_2_3action_recurrModelBrainParameters();
+            brainParameters.vectorActionSize = new int[] { 3, 3 }; // Invalid action
+            var errors = BarracudaModelParamLoader.CheckModel(model, brainParameters, new SensorComponent[] { sensor_21_20_3 });
+            Assert.Greater(errors.Count(), 0);
```

Comment:
should there be errors?

Reply:
Greater is strictly greater. There should be an error because the action size is different